tests/cpp_api_tests/rocAL_audio_unittests/rocAL_audio_unittests.cpp

```diff
@@ -0,0 +1,250 @@
+    if (argc < MIN_ARG_COUNT)
+        return -1;
+
+    int argIdx = 0;
+    const char *path = argv[++argIdx];
+    unsigned test_case = 0;
+    float sample_rate = 0.0;
+    bool downmix = false;
+    unsigned max_frames = 1;
+    unsigned max_channels = 1;
+    bool gpu = 0;
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        test_case = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        sample_rate = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        downmix = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        max_frames = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        max_channels = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        gpu = atoi(argv[++argIdx]);
+
+    int return_val = test(test_case, path, sample_rate, downmix, max_frames, max_channels, gpu);
+    return return_val;
+}
+
+int test(int test_case, const char *path, float sample_rate, int downmix, unsigned max_frames, unsigned max_channels, int gpu) {
+    int inputBatchSize = 10;
+    std::cout << ">>> test case " << test_case << std::endl;
+    std::cout << ">>> Running on " << (gpu ? "GPU" : "CPU") << std::endl;
+
+    auto handle = rocalCreate(inputBatchSize,
+                              gpu ? RocalProcessMode::ROCAL_PROCESS_GPU : RocalProcessMode::ROCAL_PROCESS_CPU, 0,
+                              1);
+
+    if (rocalGetStatus(handle) != ROCAL_OK) {
+        std::cout << "Could not create the Rocal contex\n";
+        return -1;
+    }
+
+    /*>>>>>>>>>>>>>>>> Creating Rocal parameters  <<<<<<<<<<<<<<<<*/
+
+    RocalMetaData metadata_output;
+    // MetaData reader for input file_list which has file seperated by labels
+    // if (METADATA) { // To uncomment later when meta-data reader for audio is added (PR4)
+    //     std::cout << "META DATA READER";
+    //     const char* file_list_path = "/workspace/rnnt/AMD/MIVisionX-data/rocal_data/audio_samples/audio_file_list.txt" ; // TODO: Add this as an arg in main()
+    //     metadata_output = rocalCreateFileListLabelReader(handle, path, file_list_path);
+    // }
+
+    // Decoder
+    //  RocalTensor input1; // Uncomment when augmentations are enabled
+    //  RocalTensorList non_silent_region_op; // Uncomment when NSR is introduced (PR5)
+    //  const char* file_list_path = "/media/MIVisionX-data/rocal_data/audio_samples/audio_file_list.txt" ; // Uncomment and use it when meta-data reader is introduced (PR4)
+    rocalAudioFileSourceSingleShard(handle, path, 0, 1, true, false, false, false, max_frames, max_channels, 0);
+    if (rocalGetStatus(handle) != ROCAL_OK) {
+        std::cout << "Audio source could not initialize : " << rocalGetErrorMessage(handle) << std::endl;
+        return -1;
+    }
+    /* The augmentation cases - To uncomment as each augmentation is introduced
```

Comment:
You can remove this as well and introduce when each augmentation will be added, Please check with @shobana-mcw

Reply:
done

Reply:
will add them as we progress with the prs. As it would be more confusing